ico/reader_test.go

```diff
@@ -14,27 +14,32 @@ import (
 func TestDecodeAll(t *testing.T) {
 	t.Parallel()
 	assert := assert.New(t)
-	files, _ := filepath.Glob("testdata/favicons/*.ico")
+	files, _ := filepath.Glob("testdata/*.ico")
 	for _, f := range files {
+		t.Log("testing:", f)
 		rd, err := os.Open(f)
 		assert.NoError(err, f)
 		images, err := ico.DecodeAll(rd)
-		assert.NoError(err, f)
 		rd.Close()
+		if len(images) < 1 {
```

Comment:
Here we are checking if an error should occur, and if so asserting that it does.
It would be better if, for each item, we checked for the expected outcome.
Either a table test, or a simpler "if filename matches X then assert an error".

Reply:
The latter will allow for easier adding of new testcases, so i will go for that.